apps/dav/lib/CalDAV/CalDavBackend.php

```
@@ -3161,6 +3161,17 @@ protected function getCalendarObjectId($calendarId, $uri, $calendarType):int {
 		return (int)$objectIds['id'];
 	}
 
+	/**
+	 * @throws \OCP\DB\Exception
```

I wouldn't even document this, because we don't expect an error to occur here during normal operation. the exception will only be seen when the database is down or similar, right?

---

I usually put this everywhere IDE suggests me, so that it's kept in mind when using the method, and there's some other methods in this file in the same case. 

In this specific case there's at least possibility of an exception when `$keep` is `< 0`, but I should probably move the check here and throw an `\InvalidArgumentException`.